From 83ba29d56dc6c79f14e0507ae1523d5e4ef55b29 Mon Sep 17 00:00:00 2001
From: dhivael <dhivael.git@eno.space>
Date: Sat, 11 Jan 2020 15:04:46 +0100
Subject: [PATCH 3/3] rk3399: light ROC-RK3399-PC yellow led during early boot

this is a hack, but it works for now.

(Adapted from commit 1a01021c9361c4e017cb5b032300f5555c393710)
---
 arch/arm/mach-rockchip/rk3399/rk3399.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/rk3399/rk3399.c b/arch/arm/mach-rockchip/rk3399/rk3399.c
index 863024d0710..54ee531d1e2 100644
--- a/arch/arm/mach-rockchip/rk3399/rk3399.c
+++ b/arch/arm/mach-rockchip/rk3399/rk3399.c
@@ -15,6 +15,8 @@
 #include <asm/arch-rockchip/grf_rk3399.h>
 #include <asm/arch-rockchip/hardware.h>
 #include <power/regulator.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pinctrl/rockchip.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -111,12 +113,14 @@ void board_debug_uart_init(void)
 {
 #define GRF_BASE	0xff770000
 #define GPIO0_BASE	0xff720000
+#define GPIO2_BASE	0xff780000
 #define PMUGRF_BASE	0xff320000
 	struct rk3399_grf_regs * const grf = (void *)GRF_BASE;
 #ifdef CONFIG_TARGET_CHROMEBOOK_BOB
 	struct rk3399_pmugrf_regs * const pmugrf = (void *)PMUGRF_BASE;
-	struct rockchip_gpio_regs * const gpio = (void *)GPIO0_BASE;
 #endif
+	struct rockchip_gpio_regs * const gpio = (void *)GPIO0_BASE;
+	struct rockchip_gpio_regs * const gpio2 = (void *)GPIO2_BASE;
 
 #if defined(CONFIG_DEBUG_UART_BASE) && (CONFIG_DEBUG_UART_BASE == 0xff180000)
 	/* Enable early UART0 on the RK3399 */
@@ -149,6 +153,14 @@ void board_debug_uart_init(void)
 	spl_gpio_set_pull(&pmugrf->gpio0_p, GPIO(BANK_B, 4), GPIO_PULL_NORMAL);
 #endif /* CONFIG_TARGET_CHROMEBOOK_BOB */
 
+	{
+		// set GPIO2_D3 to GPIO_ACTIVE_HIGH
+		// set GPIO2_D3 to OUTPUT
+		int mask = (1UL << RK_PD3);
+		setbits_le32(&gpio2->swport_dr, mask);
+		setbits_le32(&gpio2->swport_ddr, mask);
+	}
+
 	/* Enable early UART2 channel C on the RK3399 */
 	rk_clrsetreg(&grf->gpio4c_iomux,
 		     GRF_GPIO4C3_SEL_MASK,
-- 
2.23.1

